apm-agent-plugins/apm-spring-webflux/apm-spring-webflux-plugin/src/main/java/co/elastic/apm/agent/springwebflux/HandlerAdapterInstrumentation.java

```diff
@@ -98,31 +98,20 @@ public static Object onEnter(@Advice.Argument(0) ServerWebExchange exchange,
                                                  @Advice.Enter @Nullable Object enterTransaction,
                                                  @Advice.Return @Nullable Mono<HandlerResult> resultMono) {
 
-            if (!(enterTransaction instanceof Transaction)) {
+            if (!(enterTransaction instanceof Transaction) || resultMono == null) {
                 return resultMono;
             }
 
             Transaction transaction = (Transaction) enterTransaction;
             transaction.captureException(thrown);
 
-            if (resultMono == null) {
-                return resultMono;
-            }
-
             if (transaction.isNoop()) {
-                // when exception has been disabled within method invocation, we need to properly disable it
-                // without the need to use wrapping
+                // in transaction has been made no-op, we must still deactivate it
                 transaction.deactivate();
-                return resultMono;
             }
-
-            // in case an exception it thrown, it's too early to end transaction
-            // otherwise the status code returned isn't the expected one
-
-            return WebfluxHelper.wrapHandlerAdapter(tracer, resultMono, transaction, exchange);
```

Comment:
[for review] this wraps the request processing, which is not necessary as we instrument the "dispatcher" which includes this.